modules/alloy/src/main/ts/ephox/alloy/behaviour/replacing/ReplaceApis.ts

```diff
@@ -4,32 +4,42 @@ import { Compare, Insert, SugarElement } from '@ephox/sugar';
 import { AlloyComponent } from '../../api/component/ComponentApi';
 import { AlloySpec } from '../../api/component/SpecTypes';
 import * as Attachment from '../../api/system/Attachment';
-import * as AriaFocus from '../../aria/AriaFocus';
 import * as InternalAttachment from '../../system/InternalAttachment';
 import { Stateless } from '../common/BehaviourState';
+import { withoutReuse, withReuse } from './ReplacingAll';
 import { ReplacingConfig } from './ReplacingTypes';
 
-const set = (component: AlloyComponent, replaceConfig: ReplacingConfig, replaceState: Stateless, data: AlloySpec[]): void => {
-  // NOTE: we may want to create a behaviour which allows you to switch
-  // between predefined layouts, which would make a noop detection easier.
-  // Until then, we'll just use AriaFocus like redesigning does.
-  AriaFocus.preserve(() => {
-    const newChildren = Arr.map(data, component.getSystem().build);
-    InternalAttachment.replaceChildren(component, newChildren);
-  }, component.element);
+const virtualReplace = (component: AlloyComponent, replacee: AlloyComponent, _replaceeIndex: number, childSpec: AlloySpec) => {
+  InternalAttachment.virtualDetach(replacee);
+  const child = component.getSystem().buildOrPatch(childSpec, Optional.some(replacee.element));
+  InternalAttachment.virtualAttach(component, child);
+  component.syncComponents();
 };
 
-const insert = (component: AlloyComponent, replaceConfig: ReplacingConfig, insertion: (p: SugarElement<Node>, c: SugarElement<Node>) => void, childSpec: AlloySpec): void => {
+const insert = (component: AlloyComponent, insertion: (p: SugarElement<Node>, c: SugarElement<Node>) => void, childSpec: AlloySpec): void => {
   const child = component.getSystem().build(childSpec);
   Attachment.attachWith(component, child, insertion);
 };
 
+const replace = (component: AlloyComponent, replacee: AlloyComponent, replaceeIndex: number, childSpec: AlloySpec) => {
+  Attachment.detach(replacee);
+  insert(component, (p, c) => Insert.appendAt(p, c, replaceeIndex), childSpec);
+};
+
+const set = (component: AlloyComponent, replaceConfig: ReplacingConfig, replaceState: Stateless, data: AlloySpec[]): void => {
+  // NOTE: we may want to create a behaviour which allows you to switch
```

Comment:
Redesigning I think was removed over four years ago :) So I clearly kept these comments up-to-date.

Reply:
Removed this and the other comment that aren't relevant anymore.